packages/expo-firebase-analytics/src/Analytics.ts

```diff
@@ -0,0 +1,177 @@
+ *
+ * @param userID The user ID to ascribe to the user of this app on this device, which must be
+ *     non-empty and no more than 256 characters long. Setting userID to null removes the user ID.
+ */
+export async function setUserId(userId: string | null): Promise<void> {
+  if (!ExpoFirebaseAnalytics.setUserId) {
+    throw new UnavailabilityError('expo-firebase-analytics', 'setUserId');
+  }
+  return await ExpoFirebaseAnalytics.setUserId(userId);
+}
+
+/**
+ * Sets a user property to a given value. Up to 25 user property names are supported. Once set,
+ * user property values persist throughout the app life-cycle and across sessions.
+ *
+ * The following user property names are reserved and cannot be used:
+ *
+ * - `first_open_time`
+ * - `last_deep_link_referrer`
+ * - `user_id`
+ *
+ * @param name The name of the user property to set. Should contain 1 to 24 alphanumeric characters
+ *     or underscores and must start with an alphabetic character. The "firebase_", "google_", and
+ *     "ga_" prefixes are reserved and should not be used for user property names.
+ * @param value The value of the user property. Values can be up to 36 characters long. Setting the
+ *     value to null removes the user property.
+ */
+export async function setUserProperty(name: string, value: string): Promise<void> {
+  return await setUserProperties({ [name]: value });
+}
+
+/**
+ * Clears all analytics data for this instance from the device and resets the app instance ID.
+ */
+export async function resetAnalyticsData(): Promise<void> {
+  if (!ExpoFirebaseAnalytics.resetAnalyticsData) {
+    throw new UnavailabilityError('expo-firebase-analytics', 'resetAnalyticsData');
+  }
+  return await ExpoFirebaseAnalytics.resetAnalyticsData();
+}
+
+/**
+ * Sets multiple user properties to the supplied values.
+ *
+ * @param properties key/value set of user properties
+ */
+export async function setUserProperties(properties: { [key: string]: string }): Promise<void> {
+  if (!ExpoFirebaseAnalytics.setUserProperties) {
+    throw new UnavailabilityError('expo-firebase-analytics', 'setUserProperties');
+  }
+  return await ExpoFirebaseAnalytics.setUserProperties(properties);
+}
+
+/**
+ * Enables or disables the warning and log messages when using
+ * Firebase Analytics on the Expo client.
+ *
+ * Firebase Analytics is not available on the Expo client and therefore
+ * logs the requests to the console for development purposes. To test
+ * Firebase Analytics, create a stand-alone build or custom client.
+ * Use this function to suppress the warning and log messages.
+ *
+ * @param properties key/value set of user properties
+ */
+export function setUnavailabilityLogging(isEnabled: boolean): void {
+  if (!ExpoFirebaseAnalytics.setUnavailabilityLogging) {
+    throw new UnavailabilityError('expo-firebase-analytics', 'setUnavailabilityLogging');
+  }
+  return ExpoFirebaseAnalytics.setUnavailabilityLogging(isEnabled);
```

Comment:
```suggestion
  ExpoFirebaseAnalytics.setUnavailabilityLogging(isEnabled);
```
or set the return type to `Promise<void>`. All unimodule methods are currently async.